src/Microsoft.AspNetCore.WebHooks.Receivers/IWebHookRequestReader.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.WebHooks
+{
+    /// <summary>
+    /// Provides an abstraction for a service that parses the request body. For use in filters that execute prior to
+    /// regular model binding or in actions that read the request body after regular model binding.
+    /// </summary>
+    public interface IWebHookRequestReader
+    {
+        /// <summary>
+        /// Get an indication whether the <paramref name="request"/> is suitable to be read by
+        /// <see cref="ReadBodyAsync{TModel}"/> or a similar request entity body reader.
+        /// </summary>
+        /// <param name="request">The <see cref="HttpRequest"/> to check.</param>
+        /// <returns>
+        /// <see langword="true"/> if the <paramref name="request"/> is suitable to be read by
+        /// <see cref="ReadBodyAsync{TModel}"/>; <see langword="false"/> otherwise.
+        /// </returns>
```

Comment:
What does this mean though? What is `suitable` and `valid`

Reply:
Short answer is it means `ReadBodyAsync(...)` won't always give up and return `null`. I'll update the comments&hellip;

Reply:
No need to update the comments. I removed the method from the `interface`.